### PR TITLE
checker: check duplicate interfaces when implements for `struct`

### PR DESCRIPTION
**Pull request by Delta456**
This PR disallows duplicate interfaces when structs have explicit `implements` for interfaces.

### CODE REVIEW

**Comment by Krotki** [comment]
What happens if you use `walker.Visitor` on top of tripe `Visitor` types in test ? Will recognize it as a same type ?

**Comment by Delta456** [comment]
> What happens if you use `walker.Visitor` on top of tripe `Visitor` types in test ? Will recognize it as a same type ?

It does recognise the same type. I have formatted the file so it truncated the explicit import symbol.

File: vlib/v/checker/tests/struct_implements_interface_more_than_once_err.vv [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import v.ast.walker { Visitor }
+import v.ast { Node }
```
Comment (spytheman) [inline vlib/v/checker/tests/struct_implements_interface_more_than_once_err.vv:2]: Please use another interface from a module not from the `v.` ones.
`import hash { Hash }` for example, is much lighter to import, and will serve the same exact purpose.

Comment (spytheman) [reply vlib/v/checker/tests/struct_implements_interface_more_than_once_err.vv:2]: Or `Hash32er`, since it is smaller (it has just one method).

Comment (spytheman) [reply vlib/v/checker/tests/struct_implements_interface_more_than_once_err.vv:2]: Ok, `import io { Writer }` seems more suitable.
It has just one method, and it is public.

Comment (spytheman) [reply vlib/v/checker/tests/struct_implements_interface_more_than_once_err.vv:2]: Something like this for example checks for a lot less time:
```
CPU: 98% 0.11s	Real: 0.12s	Elapsed: 0:00.12	RAM: 34592KB	v vlib/v/checker/tests/struct_implements_interface_more_than_once_err.vv
```

compared to the version here:
```
CPU: 86% 0.31s	Real: 0.40s	Elapsed: 0:00.40	RAM: 91568KB	v vlib/v/checker/tests/struct_implements_interface_more_than_once_err.vv
```

Comment (spytheman) [reply vlib/v/checker/tests/struct_implements_interface_more_than_once_err.vv:2]: image

the underline for the `io.Writer` is wrong though

**Review by spytheman — APPROVED** [review]

**Pull request merged**